discard_test.go

```diff
@@ -17,12 +17,171 @@
 package badger
 
 import (
+	"bufio"
+	"bytes"
+	"fmt"
 	"os"
+	"strconv"
+	"strings"
+	"sync"
 	"testing"
+	"time"
 
+	"github.com/pkg/profile"
 	"github.com/stretchr/testify/require"
 )
 
+func farEnough(itrKey, key []byte) int {
+	n := len(itrKey)
+	m := len(key)
+	if m > n {
+		m = n
+	}
+
+	for i := 0; i < m; i++ {
+		if itrKey[i] != key[i] {
+			return m - i
+		}
+	}
+
+	return 0
+
+}
+
+type ByteSliceArray [][]byte
+
+// Implementing the sort.Interface for ByteSliceArray
+
+// Len returns the length of the ByteSliceArray.
+func (b ByteSliceArray) Len() int {
+	return len(b)
+}
+
+// Less compares two byte arrays at given indices and returns true if the byte array at index i is less than the byte array at index j.
+func (b ByteSliceArray) Less(i, j int) bool {
+	return bytesLessThan(b[i], b[j])
+}
+
+// Swap swaps the byte arrays at given indices.
+func (b ByteSliceArray) Swap(i, j int) {
+	b[i], b[j] = b[j], b[i]
+}
+
+// bytesLessThan compares two byte arrays lexicographically.
+func bytesLessThan(a, b []byte) bool {
+	return bytes.Compare(a, b) >= 0
+}
+
+func TestReadC(t *testing.T) {
+	t.Skip()
+	allKeysF, err := os.Open("/home/harshil/all_keys_2")
```

Comment:
Bit weird to have an explicit user path, make it a temp path or something?

Reply:
This s just a temp test for my local validation. Hence the t.Skip() before the test starts. I can't commit this as the directory being used is quite big.